lesson4/README.md

```diff
@@ -1,9 +1,93 @@
 # Lesson 4 - Data Storage App Life Cycle
 
 ##  Data Storage
+Everytime you close an app and reopen it, a new session of app is created. This means that any data that wasn't saved cannot be retrieved again. This is where persistent storage comes in. Android provides you with 5-6 options to choose from. While choosing an option, you should consider complexity of data being stored, privacy (if it can be read by the app only, the users or other apps on the device), and location (if it is local or on the cloud). We'll be mainly discussing Shared Preferences and SQLite databases. But here are many of the options available to Android:
+
+1. [Shared Preferences](https://developer.android.com/guide/topics/data/data-storage.html#pref):
+
+   Useful for writing small amount of data as key-value pairs.
+2. [Firebase](https://www.firebase.com/docs/android/quickstart.html):
```

Comment:
are we actually going to ask them to use this? I think it was mentioned last year in MP but nobody actually used it, did they?